src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt

```diff
@@ -76,4 +80,20 @@ class OrderService(val db: OrderRepository, val synqService: SynqOrderService) {
             throw ServerWebInputException("The order must contain product lines")
         }
     }
+
+    suspend fun getOrder(
+        hostName: HostName,
+        hostOrderId: String
+    ): ResponseEntity<Order> {
+        val order =
+            db.findByHostNameAndHostOrderId(hostName, hostOrderId)
+                .onErrorMap(IncorrectResultSizeDataAccessException::class.java) {
+                    TODO("Handle duplicate order ID's somehow")
+                }
```

Comment:
With change to DB so that hostName and hostOrderId are in a unique index, we avoid the issue of duplicate orders
That will of course require us to make sure that we don't duplicate orders when we create them

Reply:
This error handling has been removed, as we still get the error if it ever occurs